### ISSUE TITLE
`textDocument/references` misses reference for `initialize`

### ISSUE THREAD

**Issue by toan2406** [castwide/solargraph#551]
```ruby
class A
  def initialize() # trigger references for A#initialize
  end
end

class B < A
  def initialize()
    super()
  end
end
```

## Result from solargraph 0.44.3

![Screen Shot 2022-04-26 at 17 49 58](https://user-images.githubusercontent.com/2387158/165285235-96ed4911-7940-41aa-9c36-67e99e7e3379.png)

## Expectation

The references should include `super()` from line 8.

**Comment by castwide** [comment]
I agree, including the `super()` in references would be awesome. Adding it to the roadmap.
